util/rowDecoder/decoder.go

```diff
@@ -134,3 +136,74 @@ func (rd *RowDecoder) DecodeAndEvalRowWithMap(ctx sessionctx.Context, handle int
 	}
 	return row, nil
 }
+
+// SubstituteGenColsInDecodeColMap substitutes generated columns in every expression in decodeColMap
```

Comment:
```suggestion
// SubstituteGenColsInDecodeColMap substitutes generated columns in every expression 
```
This comment sounds really weird, maybe removing `in decodeColMap` is just better